saito-core/src/core/io/network.rs

```diff
@@ -349,6 +343,16 @@ impl Network {
             return;
         }
         let peer = peer.unwrap();
+        if !peer.rate_limiter.check(peer_index) {
```

Comment:
I think it should be while receiving keylists

Sending key lists involves broadcasting to all connected peers, If this operation is rate-limited, we might not be able to send to new peers because the rate limit has been exceeded. 

Conversely, rate limiting while receiving key lists can help us individually rate limit peers.